app/models/ability.rb

```diff
@@ -0,0 +1,36 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    if user.admin?
+      can :manage, :all
+    else
+      can :destroy, Post, author_id: user.id
+      can :destroy, Comment, author_id: user.id
+    end
```

Comment:
- [ ] you have done a great job authorizing users, however in this snippet, you are giving none admin users the ability to only delete posts and comments that they created, however, you are missing giving them the ability to update and create posts and comments and also the ability to read all, you can make your code to look like this 
```
can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
```